<a href="https://colab.research.google.com/github/SobokarNadiia/python_for_da_tasks/blob/main/hw_9_1_%D0%9E%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D1%96%D1%97_%D0%BD%D0%B0%D0%B4_%D0%B4%D0%B0%D0%BD%D0%B8%D0%BC%D0%B8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

(a + b) * c / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [4]:
11283 % 89


69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [5]:
11283 // 89

126

4. Запишіть у змінну `name` ваше імʼя.

In [7]:
name = 'Nadiia'

5. Запишіть у змінну `surname` ваше прізвище.

In [8]:
surname = 'Sobokar'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [9]:
name_surname = name + ' ' + surname
name_surname

'Nadiia Sobokar'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [10]:
name_surname.upper()

'NADIIA SOBOKAR'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [11]:
name_surname[2]

'd'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [13]:
print(surname[0], surname[-1])

S r


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [14]:
len(surname)

7

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [58]:
capital = 'Antananarivo'
capital = capital.lower()
print(len(capital))
count = 0
count = capital[0] == 'a'
count += capital[2] == 'a'
count += capital[3] == 'a'
count += capital[4] == 'a'
count += capital[5] == 'a'
count += capital[6] == 'a'
count += capital[7] == 'a'
count += capital[8] == 'a'
count += capital[9] == 'a'
count += capital[10] == 'a'
count += capital[11] == 'a'
count

12


4

Очікувана відповідь: 4

Наступні вправи можуть здатись дещо складніше - вони для найуважніших!
Якщо не виходить - не страшно. Пишіть нам в чат, ми підкажемо!

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [25]:
my_string =  "The quick brown fox jumps over the lazy dog!!!"
print(my_string.startswith('The'))
print(my_string.endswith('dog!'))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [27]:
my_string.replace(' ', '_')   # я розумію, що потрібно було спочатку split зробити, а потім їх сконкатувати якось, але я тількт так здогадалась(

'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

In [59]:
new = my_string.split(' ')
new[0] + '_' + new[1] + '_' + new[2] + '_' + new[3] + '_' + new[4] + '_' + new[5] + '_' + new[6] + '_' + new[7] + '_' + new[8]    #я тільки до такого додумалась

'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`